zproject_java.gsl

```diff
@@ -831,6 +872,14 @@ $(jni_shim_signature_c:))
     char *$(c_name)_ = (char *) (*env)->GetStringUTFChars (env, $(c_name), NULL);
 .       elsif type = "buffer"
     jbyte *$(c_name)_ = (byte *) (*env)->GetByteArrayElements (env, $(c_name), 0);
+.       elsif callback = 1
+    jclass $(c_name)_callbackClass = (*env)->GetObjectClass($(c_name));
+    jmethodID $(c_name)_method = (*env)->GetMethodID($(c_name)_callbackClass, "callback", $(cb_c_type));
+    if ($(c_name)_method == 0){
+        // Shortcut to throw a NoSuchMethodError
+        return;
+    }
+    $(type) $(c_name)_$(type) = $(type)_cb_wrapper;
```

Comment:
This $(c_name)_$(type) approach to generating a parameter name leaves a lot to be desired.
If nothing else, it's duplicated in too many places for comfort and should be part of the container's data instead.